src/pages/ImportData.js

```diff
@@ -127,66 +311,86 @@ const ImportData = () => {
     Utils.withBusyState(setIsImporting),
     withErrorReporting('Import Error')
   )(async workspace => {
     const { namespace, name } = workspace
 
     await Utils.switchCase(format,
-      ['PFB', async () => {
-        const { jobId } = await Ajax().Workspaces.workspace(namespace, name).importJob(url, 'pfb', null)
-        asyncImportJobStore.update(Utils.append({ targetWorkspace: { namespace, name }, jobId }))
-        notifyDataImportProgress(jobId)
-      }],
-      ['entitiesJson', async () => {
-        await Ajax().Workspaces.workspace(namespace, name).importJSON(url)
-        notify('success', 'Data imported successfully.', { timeout: 3000 })
-      }],
-      ['tdrexport', async () => {
-        const { jobId } = await Ajax().Workspaces.workspace(namespace, name).importJob(tdrmanifest, 'tdrexport', { tdrSyncPermissions: tdrSyncPermissions === 'true' })
-        asyncImportJobStore.update(Utils.append({ targetWorkspace: { namespace, name }, jobId }))
-        notifyDataImportProgress(jobId)
-      }],
-      ['snapshot', async () => {
-        if (!_.isEmpty(snapshots)) {
-          const responses = await Promise.allSettled(
-            _.map(({ title, id, description }) => {
-              return Ajax().Workspaces.workspace(namespace, name).importSnapshot(id, normalizeSnapshotName(title), description)
-            }, snapshots)
-          )
-
-          if (_.some({ status: 'rejected' }, responses)) {
-            const normalizedResponses = await Promise.all(_.map(async ({ status, reason }) => {
-              const reasonJson = await reason?.json()
-              const { message } = JSON.parse(reasonJson?.message || '{}')
-              return { status, message }
-            }, responses))
-            setSnapshotResponses(normalizedResponses)
-
-            // Consolidate the multiple errors into a single error message
-            const numFailures = _.flow(
-              _.filter({ status: 'rejected' }),
-              _.size
-            )(normalizedResponses)
-            throw new Error(`${numFailures} snapshot${numFailures > 1 ? 's' : ''} failed to import. See details in the "Linking to Workspace" section`)
-          }
-        } else {
-          await Ajax().Workspaces.workspace(namespace, name).importSnapshot(snapshotId, normalizeSnapshotName(snapshotName))
-          notify('success', 'Snapshot imported successfully.', { timeout: 3000 })
-        }
-      }],
-      ['catalog', async () => {
-        await Ajax().Catalog.exportDataset({ id: catalogDatasetId, workspaceId: workspace.workspaceId })
-        notify('success', 'Catalog dataset imported successfully.', { timeout: 3000 })
-      }],
+      ['PFB', importPFB(namespace, name)],
+      ['entitiesJson', importEntitiesJson(namespace, name)],
+      ['tdrexport', importTdrExport(namespace, name)],
+      ['snapshot', importSnapshot(namespace, name)],
+      ['catalog', exportCatalog(workspace.workspaceId)],
       [Utils.DEFAULT, async () => {
```

Comment:
I left Utils.DEFAULT here, because I think it is hard to name, and it doesn't hurt understanding of what options are in the switch to have it here.